tests/v2/waku_archive/test_driver_postgres.nim

```diff
@@ -27,76 +27,62 @@ suite "Postgres driver":
   const storeMessageDbUrl = "postgres://postgres:test123@localhost:5432/postgres"
 
   asyncTest "Asynchronous queries":
-    #TODO: make the test asynchronous
-    return
+    let driverRes = PostgresDriver.new(dbUrl = storeMessageDbUrl,
+                                       maxConnections = 100)
 
-    ## When
-    let driverRes = PostgresDriver.new(storeMessageDbUrl)
+    assert driverRes.isOk(), driverRes.error
 
-    ## Then
-    require:
-      driverRes.isOk()
+    let driver = driverRes.value
+    let _ = await driver.reset()
 
-    let driver: ArchiveDriver = driverRes.tryGet()
-    require:
-      not driver.isNil()
+    var futures = newSeq[Future[ArchiveDriverResult[void]]](0)
 
     let beforeSleep = now()
-    for _ in 1 .. 20:
-      discard (PostgresDriver driver).sleep(1)
+    for _ in 1 .. 100:
+      futures.add(driver.sleep(1))
 
-    require (now() - beforeSleep) < 20
+    await allFutures(futures)
 
-    ## Cleanup
-    (await driver.close()).expect("driver to close")
+    let diff = now() - beforeSleep
+    # Actually, the diff randomly goes between 1 and 2 seconds.
+    # although in theory it should spend 1s because we establish 100
+    # connections and we spawn 100 tasks that spend ~1s each.
+    require diff < 20
 
-  asyncTest "init driver and database":
+    (await driver.close()).expect("driver to close")
 
-    ## When
+  asyncTest "Init database":
     let driverRes = PostgresDriver.new(storeMessageDbUrl)
+    assert driverRes.isOk(), driverRes.error
 
-    ## Then
-    require:
-      driverRes.isOk()
-
-    let driver: ArchiveDriver = driverRes.tryGet()
-    require:
-      not driver.isNil()
+    let driver = driverRes.value
+    let _ = await driver.reset()
 
-    discard driverRes.get().reset()
-    let initRes = driverRes.get().init()
-
-    require:
-      initRes.isOk()
+    let initRes = await driver.init()
+    assert initRes.isOk(), initRes.error
 
-    ## Cleanup
     (await driver.close()).expect("driver to close")
 
-  asyncTest "insert a message":
-    ## Given
+  asyncTest "Insert a message":
     const contentTopic = "test-content-topic"
 
     let driverRes = PostgresDriver.new(storeMessageDbUrl)
+    assert driverRes.isOk(), driverRes.error
 
-    require:
-      driverRes.isOk()
+    let driver = driverRes.value
 
-    discard driverRes.get().reset()
-    discard driverRes.get().init()
+    let _ = await driver.reset()
 
-    let driver: ArchiveDriver = driverRes.tryGet()
-    require:
-      not driver.isNil()
+    let initRes = await driver.init()
+    assert initRes.isOk(), initRes.error
```

Comment:
Maybe we should follow this pattern in other test files too so we can see why the result is an error!
non-blocker

Reply:
Morning @rymnc ! After checking the code to apply this change I see that the change would be quite wide but we won't get very great investment return by doing so, as the tests don't tend to fail very often.

Therefore, I think is better to just apply this technique from now on in the upcoming tests.

Of course, if you consider it worth the effort, then I will make it for the current tests :)